test/language/expressions/exponentiation/exp-operator-precedence-unary-expression-semantics.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2016 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Rick Waldron
+esid: sec-unary-operators
+description: Exponentiation Operator expression precedence of unary operators
+info: >
+  ExponentiationExpression :
+    UnaryExpression
+    ...
+
+  UnaryExpression :
+    ...
+    `delete` UnaryExpression
+    `void` UnaryExpression
+    `typeof` UnaryExpression
+    `+` UnaryExpression
+    `-` UnaryExpression
+    `~` UnaryExpression
+    `!` UnaryExpression
+---*/
+
+assert.sameValue(-(3 ** 2), -9, "-(3 ** 2) === -9");
+assert.sameValue(+(3 ** 2), 9, "+(3 ** 2) === 9");
+assert.sameValue(~(3 ** 2), -10, "~(3 ** 2) === -10");
+assert.sameValue(!(3 ** 2), false, "!(3 ** 2) === false");
+
+
+assert.sameValue(2 ** -2, 0.25);
+
+var o = { p: 1 };
+
+assert.sameValue(2 ** delete o.p, 2, "delete o.p -> true -> ToNumber(true) -> 1");
+assert.sameValue(2 ** void 1, NaN, "void 1 -> undefined -> ToNumber(undefined) -> NaN");
+assert.sameValue(2 ** typeof 1, NaN, "typeof 1 -> 'number' -> ToNumber('number') -> NaN");
+
+var s = "2";
+var n = 2;
+
+assert.sameValue(2 ** +s, 4, "+s -> +'2' -> 2 -> ToNumber(2) -> 2");
+assert.sameValue(2 ** +n, 4, "+s -> +2 -> 2 -> ToNumber(2) -> 2");
+
+assert.sameValue(2 ** -s, 0.25, "-s -> -'2' -> -2 -> ToNumber(-2) -> -2");
+assert.sameValue(2 ** -n, 0.25, "-s -> -2 -> -2 -> ToNumber(-2) -> -2");
+
+assert.sameValue(2 ** ~s, 0.125, "~s -> ~'2' -> -3 -> ToNumber(-3) -> -3");
+assert.sameValue(2 ** ~n, 0.125, "~s -> ~2 -> -3 -> ToNumber(-3) -> -3");
+
+assert.sameValue(2 ** !s, 1, "!s -> !'2' -> false -> ToNumber(false) -> 0");
+assert.sameValue(2 ** !n, 1, "!s -> !2 -> false -> ToNumber(false) -> 0");
+
+
+var capture = [];
+
+(capture.push("left"), leftValue) ** +(capture.push("right"), rightValue);
```

Comment:
No, I accidentally deleted the definitions of those objects. Check the test just before this one